hivemq-edge/src/main/java/com/hivemq/edge/modules/adapters/params/impl/ProtocolAdapterPollingInputImpl.java

```diff
@@ -58,4 +62,14 @@ public TimeUnit getUnit() {
     public int getMaxErrorsBeforeRemoval() {
         return maxErrorsBeforeRemoval;
     }
+
+    @Override
+    public void close() {
```

Comment:
Is the function being called? I couldn't find it.

Reply:
Yup - it's called by the monitoring job on shutdown. FWIW - I don't like the pattern of the runnable being the Input object - we should refactor this at some point. The small issue is the interfaces need to be appropriate for both async and sync patterns - since we don't know what the underlying connection API will support.

`if(future.cancel(false)){
                    pollingJob.getInput().close();
                }`